packages/rrweb/src/replay/index.ts

```diff
@@ -402,9 +415,32 @@ export class Replayer {
         );
       }, 1);
     }
-    if (this.service.state.context.events.find(indicatesTouchDevice)) {
-      this.mouse.classList.add('touch-device');
-    }
+    this.service.state.context.events.forEach((e: eventWithTime) => {
+      if (indicatesTouchDevice(e)) {
+        const d = e.data as incrementalData;
+        const pointerId =
+          'pointerId' in d && d.pointerId !== undefined
+            ? d.pointerId
+            : Object.keys(this.pointers).length;
+
+        if (!this.pointers[pointerId]) {
+          this.createNewTouch(pointerId);
+        }
```

Comment:
this bit could be extracted and put into a separate function - it repeats a few times